crypto/multisig/src/lib.rs

```diff
@@ -69,30 +69,72 @@ impl<
     }
 }
 
-/// A set of M-out-of-N public keys.
+/// A set of M-out-of-N signer identities, where a signer identity can be either
+/// a public key or a nested M-out-of-N set of identities.
 #[derive(
     Clone, Deserialize, Digestible, Eq, Hash, Message, Ord, PartialEq, PartialOrd, Serialize,
 )]
 #[serde(bound = "")]
 pub struct SignerSet<P: Default + PublicKey + Message> {
-    /// List of potential signers.
+    /// List of potential individual signers.
     #[prost(message, repeated, tag = "1")]
-    signers: Vec<P>,
-
-    /// Minimum number of signers required.
+    #[digestible(name = "signers")]
+    individual_signers: Vec<P>,
+
+    /// List of potential sets of signers. This allows us to declare a nested
+    /// multi-signature scheme. Ideally the two arrays in this struct would be
+    /// combined into a single one that holds an enum of (IndividualSigner,
+    /// MultiSignerSet), but since this feature was added after we went live
+    /// and instances of SignerSet made their way into the ledger, it is not
+    /// possible to change the array type without breaking backwards
+    /// compatibility. This is also the reason the tag numbers in the struct
+    /// are not sequential.
+    #[prost(message, repeated, tag = "3")]
+    multi_signers: Vec<SignerSet<P>>,
+
+    /// Minimum number of signers required. The potential signers are the union
+    /// of `individual_signers` and `multi_signers`.
+    /// This implies that the upper limit (total number of possible signers) is
+    /// `individual_signers.len() + multi_signers.len()`.
     #[prost(uint32, tag = "2")]
     threshold: u32,
 }
 
 impl<P: Default + PublicKey + Message> SignerSet<P> {
     /// Construct a new `SignerSet` from a list of public keys and threshold.
-    pub fn new(signers: Vec<P>, threshold: u32) -> Self {
-        Self { signers, threshold }
+    pub fn new(individual_signers: Vec<P>, threshold: u32) -> Self {
+        Self::new_with_multi(individual_signers, Vec::new(), threshold)
```

Comment:
Optional/Equivalent: Use `vec!`
```suggestion
        Self::new_with_multi(individual_signers, vec![], threshold)
```